lib/Service/ReportService.php

```diff
@@ -0,0 +1,230 @@
+<?php
+/**
+ * Copyright (C) 2019 Xibo Signage Ltd
+ *
+ * Xibo - Digital Signage - http://www.xibo.org.uk
+ *
+ * This file is part of Xibo.
+ *
+ * Xibo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Xibo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Xibo\Service;
+
+use Xibo\Exception\NotFoundException;
+use Xibo\Factory\DisplayFactory;
+use Xibo\Factory\LayoutFactory;
+use Xibo\Factory\MediaFactory;
+use Xibo\Factory\SavedReportFactory;
+use Xibo\Storage\StorageServiceInterface;
+use Xibo\Storage\TimeSeriesStoreInterface;
+
+/**
+ * Class ReportScheduleService
+ * @package Xibo\Service
+ */
+class ReportService implements ReportServiceInterface
+{
+
+    /**
+     * @var \Xibo\Helper\ApplicationState
+     */
+    private $state;
+    /**
+     * @var StorageServiceInterface
+     */
+    private $store;
+
+    /**
+     * @var TimeSeriesStoreInterface
+     */
+    private $timeSeriesStore;
+
+    /**
+     * @var LogServiceInterface
+     */
+    private $log;
+
+    /**
+     * @var ConfigServiceInterface
+     */
+    private $config;
+
+    /**
+     * @var DateServiceInterface
+     */
+    private $date;
+
+    /**
+     * @var SanitizerServiceInterface
+     */
+    private $sanitizer;
+
+    /**
+     * @var SavedReportFactory
+     */
+    private $savedReportFactory;
+
+    /**
+     * @var DisplayFactory
+     */
+    private $displayFactory;
```

Comment:
Again it feels like these factories are specific to our report - not generic to the service - see prior comment